### PR TITLE
Add explanation about read/write data race

### PR DESCRIPTION
**Pull request by ayuishii**
Discussion: https://github.com/WICG/cookie-store/issues/93

### CODE REVIEW

File: explainer.md [hunk 1]
```diff
@@ -572,6 +572,24 @@ const cookie = await cookieStore.get('session-id');
 console.log(cookie);
 ```
 
+#### Read / Write Race
+
+Cookie store operations are not an atomic read-modify-write transaction. Failure to await during an operation can cause unexpected results as the following.
+
+```javascript
+function increment_cookie(name) {
```
Comment (pwnall) [inline explainer.md:580]: `async function` because you use await

`incrementCookie` because JavaScript

File: explainer.md [hunk 2]
```diff
@@ -572,6 +572,24 @@ const cookie = await cookieStore.get('session-id');
...
+  const cookie = await cookieStore.get(name);
+  cookie.value = parseInt(cookie.value) + 1;
+  await cookieStore.set(cookie);
+  return cookie;
+}
+
+await cookieStore.set('cookie-name', 1);
+const cookie1 = increment_cookie('cookie-name');  // Missing "await".
```
Comment (pwnall) [inline explainer.md:588]: The values here are cookie1_promise and cookie2_promise

How about

```js
await Promise.all([
  incrementCookie('cookie-name'),
  incrementCookie('cookie-name'),
]);
const cookie = await cookieStore.get('cookie-name');
console.log(cookie.value);  // will likely be '2' not '3'
```

File: explainer.md [hunk 3]
```diff
@@ -572,6 +572,24 @@ const cookie = await cookieStore.get('session-id');
 console.log(cookie);
 ```
 
+#### Read / Write Race
+
+Cookie store operations are not an atomic read-modify-write transaction. Failure to await during an operation can cause unexpected results as the following.
```
Comment (pwnall) [inline explainer.md:577]: The ability of passing the return value of `cookieStore.set()` to `cookieStore.get()` may lead to code that reads like an atomic read-modify-write operation, when it reality it's not.

**Review by pwnall — CHANGES_REQUESTED** [review]
Thank you very much for taking this on!

Here are a couple of suggestions for changes.

I found that testing my examples in DevTools helps remove small JS bugs, leading to more credible explainers and specs.

**Comment by ayuishii** [comment]
Thanks for the review! And pointing out all the issues.
Will be better about verifying in the future...

**Review by pwnall — APPROVED** [review]
LGTM, thank you for the quick revisions!

**Pull request merged**